src/dos/drive_overlay.cpp

```diff
@@ -487,9 +491,23 @@ std::unique_ptr<DOS_File> Overlay_Drive::FileOpen(const char* name, uint8_t flag
 			LOG_MSG("overlay file opened %s", newname);
 		}
 
-		auto file = std::make_unique<OverlayFile>(
-		        name, newname, file_handle, overlaydir, IsReadOnly());
-		file->flags          = flags;
+		DosDateTime dos_time   = {};
+		const auto cache_entry = timestamp_cache.find(newname);
+		if (cache_entry == timestamp_cache.end()) {
+			dos_time = get_dos_file_time(file_handle);
+			timestamp_cache[newname] = dos_time;
+		} else {
+			dos_time = cache_entry->second;
+		}
+
+		auto file            = std::make_unique<OverlayFile>(name,
+                                                          newname,
+                                                          file_handle,
+                                                          overlaydir,
+                                                          IsReadOnly(),
+                                                          weak_from_this(),
+                                                          dos_time,
+                                                          flags);
```

Comment:
formatting

Reply:
You can blame clang-format for that.  It's not perfect.  Fixed it manually though.